src/__tests__/Transposit.test.ts

```diff
@@ -269,94 +270,142 @@ describe("Transposit", () => {
+            JSON.stringify({
+              status: "SUCCESS",
+              requestId: "12345",
+              result: {
+                results: ["hello", "world"],
+              },
+            } as EndRequestLog),
+          ),
         ),
-      ),
-    );
+      );
 
-    const results: EndRequestLog = await transposit.run("hello_world");
+      const response: OperationResponse<string> = await transposit.run<string>(
+        "hello_world",
+      );
 
-    expect(results.result.results).toEqual(["hello", "world"]);
-    expect(window.fetch as jest.Mock).toHaveBeenCalledWith(
-      "https://arbys-beef-xyz12.transposit.io/api/v1/execute/hello_world",
-      {
-        method: "POST",
-        headers: {
-          "Content-Type": "application/json",
-          Authorization:
-            "Bearer eyJhbGciOiJub25lIn0=.eyJpc3MiOiJodHRwczovL2FyYnlzLWJlZWYteHl6MTIudHJhbnNwb3NpdC5pbyIsInN1YiI6Imdvb2dsZXxzYW5kZXJzQGtlcm5lbC5jb20iLCJleHAiOjE1MjI1MTQ1MTksImlhdCI6MTUyMTk5NjExOX0=.",
+      expect(response.requestId).toEqual("12345");
+      expect(response.results).toEqual(["hello", "world"]);
+      expect(response.value).toEqual("hello");
+
+      expect(window.fetch as jest.Mock).toHaveBeenCalledWith(
+        "https://arbys-beef-xyz12.transposit.io/api/v1/execute/hello_world",
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+            Authorization:
+              "Bearer eyJhbGciOiJub25lIn0=.eyJpc3MiOiJodHRwczovL2FyYnlzLWJlZWYteHl6MTIudHJhbnNwb3NpdC5pbyIsInN1YiI6Imdvb2dsZXxzYW5kZXJzQGtlcm5lbC5jb20iLCJleHAiOjE1MjI1MTQ1MTksImlhdCI6MTUyMTk5NjExOX0=.",
+          },
+          body: '{"parameters":{}}',
         },
-        body: '{"parameters":{}}',
-      },
-    );
-  });
+      );
+    });
 
-  it("run an operation and throws errors", async () => {
-    expect.assertions(2);
+    it("run an operation and throws APIError", async () => {
+      expect.assertions(2);
 
-    makeSignedIn(accessToken);
-    const transposit: Transposit = new Transposit(BACKEND_ORIGIN);
+      makeSignedIn(accessToken);
+      const transposit: Transposit = new Transposit(BACKEND_ORIGIN);
 
-    (window.fetch as jest.Mock).mockReturnValueOnce(
-      Promise.resolve(
-        new Response(null, { status: 400, statusText: "Bad Request" }),
-      ),
-    );
-    try {
-      await transposit.run("hello_world");
-    } catch (e) {
-      expect(e).toBeInstanceOf(APIError);
-      expect(e.message).toEqual(INTERNAL_ERROR_MESSAGE);
-    }
+      (window.fetch as jest.Mock).mockReturnValueOnce(
+        Promise.resolve(
+          new Response(null, { status: 400, statusText: "Bad Request" }),
+        ),
+      );
+      try {
+        await transposit.run<string>("hello_world");
+      } catch (e) {
+        expect(e).toBeInstanceOf(APIError);
+        expect(e.message).toEqual(INTERNAL_ERROR_MESSAGE);
+      }
+    });
+
+    const badStatuses = ["ERROR", "CANCELLED", "TIMEOUT"];
```

Comment:
nit: maybe also include `RESOURCELIMITEXCEEDED`? (the EndRequestLog interface is a little outdated)